garden_ai/client.py

```diff
@@ -362,6 +364,126 @@ def register_pipeline(self, pipeline: Pipeline, container_uuid: str) -> str:
         local_data.put_local_pipeline(registered)
         return func_uuid
 
+    def create_minimal_step(
+        self,
+        *,
+        model_name: str,
+        local_path: str,
+        flavor: str,
+        input_type: type,
+        output_type: type,
+        requirements_file: str = "requirements.txt",
+    ) -> Step:
+        with open(requirements_file, "r") as f:
+            pip_reqs = [line.strip() for line in f.readlines()]
+
+        local_model = LocalModel(
+            model_name=model_name,
+            flavor=flavor,
+            extra_pip_requirements=pip_reqs,
+            local_path=local_path,
+            user_email=self.get_email(),
+        )
+
+        registered_model = self.register_model(local_model)
+
+        @step
+        def run_inference(
+            input_arg: input_type, model=Model(registered_model.full_name)
+        ) -> output_type:
+            return model.predict(input_arg)
+
+        return run_inference
+
+    def publish_minimal_pipeline(
+        self,
+        _step: Step,
+        garden_doi: str,
+        *,
+        authors: List[str],
+        tags: List[str],
+        title: str = None,
+        description: str = None,
+        requirements_file: str = "requirements.txt",
+    ) -> Garden:
+        full_name = _step.model_full_names[0]
+        pipeline = self.create_pipeline(
+            authors,
+            title or f"Inference on {full_name}",
+            short_name=full_name[full_name.index("/") + 1 :],
```

Comment:
nitpick: the user is providing us the short name as part of the simple publishing flow, but here we have to take apart a string to figure it out again